modules/core/src/main/scala/gem/Observation.scala

```diff
@@ -32,10 +30,10 @@ object Observation {
 
   }
 
-  implicit def ObservationTraverse[T]: Traverse[Observation[?]] =
-    new Traverse[Observation[?]] {
-      def traverseImpl[G[_]: Applicative, A, B](fa: Observation[A])(f: A => G[B]): G[Observation[B]] =
-        fa.steps.traverse(f).map(ss => fa.copy(steps = ss))
+  implicit val ObservationBitraverse: Bitraverse[Observation] =
```

Comment:
`Observation[A, B]` is a functor in two ways … over the `A` and the `B`, so it can be a `Bifunctor` with `leftMap` and `rightMap`.

Even better, it's a *traversable* functor in two ways, which means that in addition to `.map` you can also `.traverse` in two ways, or `.bitraverse` to do both at once. Here we take an `Observation` whose children are (fake) database ids, and then we `bitraverse` to turn it into a `ConnectionIO` that materializes both the static and dynamic components.

```scala
// An observation with database references
val o: Observation[Int, Int] =
  Observation(Observation.Id.fromString("GS-2011A-Q-3-12").get, "Test", 123, List(456, 789))

// Programs to resolve a reference into a fetched value
def loadStaticConfig(id: Int): ConnectionIO[StaticConfig] = ???
def loadDynamicConfig(id: Int): ConnectionIO[DynamicConfig] = ???

// The result type of bitraversal
scala> :t o.bitraverse(loadStaticConfig, loadDynamicConfig)
ConnectionIO[Observation[StaticConfig, DynamicConfig]]

// Just do the left side
scala> :t Bitraverse[Observation].leftTraverse.traverse(o)(loadStaticConfig)
doobie.imports.ConnectionIO[gem.Observation[gem.config.StaticConfig,Int]]
```

There's no syntax for `.leftTraverse`/`.rightTraverse` so we might add that. If we run into a type with more than two kinds of foreign keys we'll need to define something more general.